### DISCUSSION TITLE
Question about rejection of my PR's

### DISCUSSION THREAD

**Discussion by Miniontoby** [pelican-dev/panel#111] [General]
I have a question about rejection of my PR's, since I am not able to comment on them...

It is about #101 and #102 

I said "This would allow __later on__ to support running in an subfolder"
I didn't say that it should run in a subfolder!

The main reason why I created the PR is because the fact that the app is still using fixed static resource url's that don't change with the URL of the app!

The changes themselfs do not effect the app at all, only when you run in a subfolder.

And to also say it: "I have already been able to make it fully work in a subfolder", so it is possible, tho. 

@lancepioch

**Comment by Miniontoby** [comment]
Please at least apply the
```patch
diff --git a/bootstrap/app.php b/bootstrap/app.php
index 21b41272f..fa0504e38 100644
--- a/bootstrap/app.php
+++ b/bootstrap/app.php
@@ -16,7 +16,7 @@ return Application::configure(basePath: dirname(__DIR__))
         health: '/up',
     )
     ->withMiddleware(function (Middleware $middleware) {
-        $middleware->redirectGuestsTo(fn () => route('login'));
+        $middleware->redirectGuestsTo(fn () => route('auth.login'));

         $middleware->web(\App\Http\Middleware\LanguageMiddleware::class);

```

Since else it will give you an error when trying to contact the API when you're not logged in!

**Reply by Miniontoby** [reply]
@lancepioch Please give me an explanation!

**Comment by lancepioch** [comment]
I absolutely don’t want anybody to run this in a subdirectory, ever. You are free to make a plugin that accomplishes this down the road, but we cannot support that.

As for this specific bug, thanks for pointing it out, I will look at that this week, thanks!

In the future, please double check with the Pelican Team for any pull requests that aren’t a bug fix.

**Reply by Miniontoby** [reply]
> I absolutely don’t want anybody to run this in a subdirectory, ever. You are free to make a plugin that accomplishes this down the road, but we cannot support that.

I said: "The main reason why I created the PR is because the fact that the app is still using fixed static resource url's that don't change with the URL of the app!"

It wasn't the idea for running it in a subfolder at first!

But yeah, I do still want to run in subfolder, since in my local development setup on my raspberry pi, I cannot make subdomains (*No, I do not require help!*).

*And even WHMCS is able to run in a subfolder, so why wouldn't this be, tho...*

But sure. I will make a plugin. But still, it would really help when the static urls are surrounded by the `url()` function, which wouldn't change anything to a normal panel running in the root directory! Most of the assets used are already wrapped by a function, like for the theme (which doesn't really make sense for some files, tho...) so why wouldn't the rest also be wrapped in a function (the `url()` function)
